st2common/st2common/validators/api/reactor.py

```diff
@@ -15,7 +15,11 @@
 
 import six
 
+from apscheduler.triggers.cron import CronTrigger
```

Comment:
> On top of that, we already perform validation here so I don't see how performing some additional validation breaks the abstraction. For a more specific validation (on top of what JSON schema provides) you need trigger-specific information and I see no issues with putting this here.

The validation being performed here is generic based on the schema information provided. This  code nothing about the specifics of any trigger

There is a very good reason that we have implemented St2Timer as an encapsulated class and placed it in a module. The explicit reason is that only that code knows anything about the specifics of a Timer. We have gone out of our way to build out the abstraction of a trigger-type, trigger and post that onto a message bus which any interested party can consume. The reason we have built out these abstractions is so that there is a decoupled system and by adding this special validation here we are completely laying to waste the basis for these abstractions. I still see tremendous value in these abstractions. How can someone logically explain that this code is aware of `core.st2.CronTimer`? Why would a piece of code that knows nothing about the specific implementation details of any trigger suddenly start effectively "down-casting" to a specific trigger and perform special checks.

I personally see an issue with this and am not agreeable to this validation being performed in this code path.